test/controllers/registrations_controller_test.rb

```diff
@@ -116,4 +116,79 @@ class RegistrationsControllerTest < ActionController::TestCase
 
     assert_redirected_to course_registrations_path(@cs301)
   end
+
+  def attempt_register_user(user, role)
+    post :create, params: {
+        course_id: @cs101.id,
+        registration: {
+            username: user.username,
+            role: role
+        },
+        new_sections: [@section.crn.to_s]
+    }
+  end
+
+  test "graders cannot create registrations" do
+    mike_reg = Registration.create(course: @cs101,
+                                   user: @mike,
+                                   role: "grader",
+                                   show_in_lists: false)
+    mike_reg.save!
+    sign_in @mike
+    attempt_register_user @john, "student"
+    assert_redirected_to root_path
+    assert_match "Must be an admin, professor or assistant.", flash[:alert]
```

Comment:
Worth confirming here that John's registrations haven't changed?  (I.e., not just that the UI looks right, but the database is unchanged.)